actors/evm/src/interpreter/precompiles/fvm.rs

```diff
@@ -184,13 +184,15 @@ pub(super) fn call_actor<RT: Runtime>(
     let flags = SendFlags::from_bits(flags).ok_or(PrecompileError::InvalidInput)?;
 
     let codec: u64 = input_params.next_param_padded()?;
-    // TODO only CBOR for now
-    if codec != fvm_ipld_encoding::DAG_CBOR {
-        return Err(PrecompileError::InvalidInput);
-    }
 
     let address_size = input_params.next_param_padded::<u32>()? as usize;
     let send_data_size = input_params.next_param_padded::<u32>()? as usize;
+    // TODO only CBOR or "nothing" for now
+    if (codec != fvm_ipld_encoding::DAG_CBOR || send_data_size == 0)
+        && (codec != 0 || send_data_size != 0)
+    {
+        return Err(PrecompileError::InvalidInput);
+    }
```

Comment:
I would parse the input data first, then:

```suggestion
    let params = match codec {
        fvm_ipld_encoding::DAG_CBOR => Some(IpldBlock { codec, data: input_data.into() }),
        0 if input_data.is_empty() => None,
        _ => return Err(PrecompileError::InvalidInput),
    };
```

Reply:
Basically, we should never pass the "0" codec to the system. We're using it here because it's a convenient way for a solidity program to say "no parameters".

Reply:
Okay, that makes sense.